performance/admin.py

```diff
@@ -1,9 +1,168 @@
+                department=department_obj,
+                source_id=obj_id,
+
+                q1_target=indicator_data[11],
+                q1_actual_output=indicator_data[12],
+                q1_deviation_reason=indicator_data[13],
+                q1_corrective_action=indicator_data[14],
+                q1_national_comments=indicator_data[16],
+                q1_otp_comments=indicator_data[15],
+                q1_dpme_coordinator_comments='',
+                q1_treasury_comments='',
+
+                q2_target=indicator_data[17],
+                q2_actual_output=indicator_data[18],
+                q2_deviation_reason=indicator_data[19],
+                q2_corrective_action=indicator_data[20],
+                q2_national_comments=indicator_data[22],
+                q2_otp_comments=indicator_data[21],
+                q2_dpme_coordinator_comments='',
+                q2_treasury_comments='',
+
+                q3_target=indicator_data[23],
+                q3_actual_output=indicator_data[24],
+                q3_deviation_reason=indicator_data[25],
+                q3_corrective_action=indicator_data[26],
+                q3_national_comments=indicator_data[28],
+                q3_otp_comments=indicator_data[27],
+                q3_dpme_coordinator_comments='',
+                q3_treasury_comments='',
+
+                q4_target=indicator_data[29],
+                q4_actual_output=indicator_data[30],
+                q4_deviation_reason=indicator_data[31],
+                q4_corrective_action=indicator_data[32],
+                q4_national_comments=indicator_data[34],
+                q4_otp_comments=indicator_data[33],
+                q4_dpme_coordinator_comments='',
+                q4_treasury_comments='',
+
+                annual_target=indicator_data[35],
+                annual_aggregate_output='',
+                annual_pre_audit_output=indicator_data[37],
+                annual_deviation_reason=indicator_data[38],
+                annual_corrective_action=indicator_data[39],
+                annual_otp_comments=indicator_data[40],
+                annual_national_comments=indicator_data[41],
+                annual_dpme_coordincator_comments='',
+                annual_treasury_comments='',
+                annual_audited_output=indicator_data[42],
+
+                sector=indicator_data[0],
+                programme_name=indicator_data[2],
+                subprogramme_name=indicator_data[3],
+                frequency=indicator_data[5],
+                type=indicator_data[7],
+                subtype=indicator_data[8],
+                mtsf_outcome=indicator_data[9],
+                cluster=indicator_data[10],
+                uid=indicator_data[43],
+            )
+            num_imported = num_imported + 1
+        elif department_name not in not_matching_departments:
+            not_matching_departments.append(department_name)
+
+    obj_to_update = models.EQPRSFileUpload.objects.get(id=obj_id)
```

Comment:
is this what you're trying to achieve here? https://docs.djangoproject.com/en/4.1/ref/models/instances/#django.db.models.Model.refresh_from_db

Reply:
if I pass the object itself to the async task and try to update it there, I get connection already closed error, which makes sense. that's why I pass the id and find the object with id. I did this also because I thought this would make things easier in case we would call that async function from some other function in the future